### PR TITLE
rush: fix build

### PR DESCRIPTION
**Pull request by xeji**
###### Motivation for this change

An incompatibility with our latest gnulib, see [Debian bug 853649](https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=853649). Fortunately, Debian has a patch.

/cc ZHF #36453 

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `build-use-sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

File: pkgs/shells/rush/default.nix [hunk 1]
```diff
@@ -8,7 +8,12 @@ stdenv.mkDerivation rec {
     sha256 = "1vxdb81ify4xcyygh86250pi50krb16dkj42i5ii4ns3araiwckz";
   };
 
-  patches = [ ./fix-format-security-error.patch ];
+  patches = [ ./fix-format-security-error.patch
+    (fetchpatch {
+      url = "https://sources.debian.org/data/main/r/rush/1.8+dfsg-1.1/debian/patches/intprops.patch";
```
Comment (fpletz) [inline pkgs/shells/rush/default.nix]: Unfortunately this is not a stable URL. When Debian releases new versions this version will go away. As the packaging it not in version control either, we should copy the patch into nixpkgs.

**Comment by xeji** [comment]
Debian patch copied into nixpkgs and squashed commits

**Comment by xeji** [comment]
@GrahamcOfBorg eval

**Comment by xeji** [comment]
@GrahamcOfBorg eval

**Comment by fpletz** [comment]
@GrahamcOfBorg build rush

**Comment by fpletz** [comment]
Evaluation doesn't fail for me.

**Comment by volth** [comment]
> An incompatibility with our latest gnulib

Why not just update ```gnulib```?
"our latest gnulib" is more than 2 years old

**Comment by xeji** [comment]
Sorry, I misread the Debian bug report. Actually, rush bundles some code from a newer version of gnulib, which caused the build error with gcc7. It does not use our old version of gnulib at all...

**Pull request merged**